src/EnergyPlus/ZoneEquipmentManager.cc

```diff
@@ -867,366 +855,202 @@ void SetUpZoneSizingArrays(EnergyPlusData &state)
-                calcZoneSizing.DesCoolMinAirFlowPerArea = zoneSizingInput.DesCoolMinAirFlowPerArea;
-                calcZoneSizing.DesCoolMinAirFlow = zoneSizingInput.DesCoolMinAirFlow;
-                calcZoneSizing.DesCoolMinAirFlowFrac = zoneSizingInput.DesCoolMinAirFlowFrac;
-                calcZoneSizing.InpDesHeatAirFlow = zoneSizingInput.DesHeatAirFlow;
-                calcZoneSizing.DesHeatMaxAirFlowPerArea = zoneSizingInput.DesHeatMaxAirFlowPerArea;
-                calcZoneSizing.DesHeatMaxAirFlow = zoneSizingInput.DesHeatMaxAirFlow;
-                calcZoneSizing.DesHeatMaxAirFlowFrac = zoneSizingInput.DesHeatMaxAirFlowFrac;
-                calcZoneSizing.HeatSizingFactor = zoneSizingInput.HeatSizingFactor;
-                calcZoneSizing.CoolSizingFactor = zoneSizingInput.CoolSizingFactor;
-                calcZoneSizing.AccountForDOAS = zoneSizingInput.AccountForDOAS;
-                calcZoneSizing.DOASControlStrategy = zoneSizingInput.DOASControlStrategy;
-                calcZoneSizing.DOASLowSetpoint = zoneSizingInput.DOASLowSetpoint;
-                calcZoneSizing.DOASHighSetpoint = zoneSizingInput.DOASHighSetpoint;
-                calcZoneSizing.zoneSizingMethod = zoneSizingInput.zoneSizingMethod;
-                calcZoneSizing.zoneLatentSizing = zoneSizingInput.zoneLatentSizing;
-                calcZoneSizing.zoneRHDehumidifySetPoint = zoneSizingInput.zoneRHDehumidifySetPoint;
-                calcZoneSizing.zoneRHHumidifySetPoint = zoneSizingInput.zoneRHHumidifySetPoint;
-                calcZoneSizing.zoneRHDehumidifySchIndex = zoneSizingInput.zoneRHDehumidifySchIndex;
-                calcZoneSizing.zoneRHHumidifySchIndex = zoneSizingInput.zoneRHHumidifySchIndex;
-                calcZoneSizing.ZnLatCoolDgnSAMethod = zoneSizingInput.ZnLatCoolDgnSAMethod;
-                calcZoneSizing.LatentCoolDesHumRat = zoneSizingInput.LatentCoolDesHumRat;
-                calcZoneSizing.CoolDesHumRatDiff = zoneSizingInput.CoolDesHumRatDiff;
-                calcZoneSizing.ZnLatHeatDgnSAMethod = zoneSizingInput.ZnLatHeatDgnSAMethod;
-                calcZoneSizing.LatentHeatDesHumRat = zoneSizingInput.LatentHeatDesHumRat;
-                calcZoneSizing.HeatDesHumRatDiff = zoneSizingInput.HeatDesHumRatDiff;
-            }
+
+            if (DesDayNum == 1 && ZoneSizNum == 0) { // LKL I think this is sufficient for warning -- no need for array
+                if (!state.dataGlobal->isPulseZoneSizing) {
+                    ShowWarningError(
+                        state,
+                        format("SetUpZoneSizingArrays: Sizing for Zone=\"{}\" will use Sizing:Zone specifications listed for Zone=\"{}\".",
+                               state.dataZoneEquip->ZoneEquipConfig(CtrlZoneNum).ZoneName,
+                               zoneSizingInput.ZoneName));
+                }
+            }
+
+            zoneSizing.ZnCoolDgnSAMethod = zoneSizingInput.ZnCoolDgnSAMethod;
```

Comment:
Are these the same `struct`? You can just define a copy constructor and call it.

Reply:
No, but I suspect inheritance may help here on a future pass. The ZoneSizingInput array is what the user provides in the Sizing:Zone object. The ZoneSizing (and CalcZoneSizing) are arrays that hold that information and a lot more. So ZoneSizingInput is a subset of ZoneSizing.

Reply:
Yes, that is a very simple (but useful) use of inheritance where you are just extending the struct with more fields.

Reply:
One more thing I will point out. There is a lot of parallel construction here between heating and cooling design variables. And elsewhere between humidification and dehumidification. The variables basically come in pairs. There are two ways to take advantage of this and the best way depends on the usage. One way is "horizontally", i.e., make each variable into an array of two items, but this is kind of stupid when you only have two items. A better way is to take advantage of the redundancy "vertically" by collecting one copy of the variables into a `struct` and then having two copies of the `struct`, one called `heat` and one called `cool`. So you could do `zoneSizing.cool.DesTemp = zoneSizingInput.cool.DesTemp;` or `zoneSizing.cool = zoneSizingInput.cool;` if you have a copy constructor. Or reuse the same calculations for heating and cooling by passing them different `structs`.

FWIW Embedding one struct in another does not slow down accesses to the "inner" struct, the compiler "flattens" everything, i.e., the code reads `zoneSizing.cool.DesTemp` directly, it knows where `DesTemp` is relative to `zoneSizing`, it doesn't access `cool` first and then `DesTemp`. Another reason that references are bad, they make references to chains of pointed-to objects (which are slow) look like references to embedded structs (which are essentially free).

Reply:
Another comment. Is all this copying even necessary? Can you just `std::move` the `calcZoneSizing` array into the `zoneArray`? Or swap the pointers to them so that you are `calcZoneSizing` becomes the new `zoneSizing`? Do you actually need two parallel copies of these structures?

Reply:
As a reminder, I am still wondering what the best path forward would be.

Reply:
I think there are multiple questions here: 

* how should these structs be organized relative to one another
* is there a better organization of the structs?
* is all of this struct copying necessary?
* are these giant arrays necessary?

And multiple options for each. I don't know a priori what the "best" path forward is. It's hard (for me at least) to know how something will work before we try it. I suggest we go with the simplest refactoring first (i.e., the one that will have the least impact on the code in general) and see what that gets us and what we may still want to achieve after that.

The simplest thing to do first is to create a `struct` hierarchy, i.e., to define `ZoneSizingData` as derived from `ZoneInputSizingData`. Combined with overloading `operator=` that should reduce the code somewhat and let us see other opportunities.